modules/consensus/validtransaction.go

```diff
@@ -133,9 +158,22 @@ func validStorageProofs(tx *bolt.Tx, t types.Transaction) error {
 			segmentIndex,
 			fc.FileMerkleRoot,
 		)
-		if !verified {
+		if !verified && fc.FileSize > 0 {
```

Comment:
The code is functionally correct, but it's confusing to see a `HARDFORK` block as the last block in a function.

To build on top of what @lukechampine said:
>hardfork code should always be `blockHeight < threshold`. The current "correct" behavior should be in the main path, outside `if` statements.

I would add to that and say that a `HARDFORK` block should come before the "correct" behavior. If the `if !verified && fc.FileSize > 0 {` block is the "correct" behavior, it would go after the `HARDFORK` block.

For this particular case both orderings are functionally equivalent because we are adding a condition, not removing a condition. For example, if the hardfork was to remove a condition the `HARDFORK` block would need to come before the "correct" behavior. Consider this example
```go
// HARDFORK
if (build.Release == "standard" && blockHeight(tx) < 100e3) || (build.Release == "testing" && blockHeight(tx) < 10) {
    if !verified && SOME_OLD_CONDITION {
        return errInvalidStorageProof
    }
}

if !verified { // Note that SOME_OLD_CONDITION was removed from the "correct" behavior.
    return errInvalidStorageProof
}
```

Swapping the order of these blocks would not result in the desired behavior.